common/djangoapps/util/file.py

```diff
@@ -156,6 +156,7 @@ def generate_lines(self):
         replacing CR and CRLF with LF.
         """
         buf = self.original_file.read(self.buffer_size)
+        buf = buf.decode('utf-8') if six.PY3 and isinstance(buf, bytes) else buf
```

Comment:
This can break if the end of the buffer lands in the middle of a multi-byte character.  At a glance, this seems kind of tricky to get correct; we may need to replace this whole method with a new implementation utilizing the classes in the `io` standard library module that handle streaming Unicode text with universal newline support.

Reply:
@jmbowman do we want to put this change on hold for now?

Reply:
It looks like we only use this class for CSV parsing in a couple of places, and in one of them we already updated the code to just use the standard library instead under Python 3 (since its CSV parser gained proper Unicode support).  Maybe we should completely stop using this under Python 3 and skip the tests when running that version?

Already refactored: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor/views/api.py#L1408-L1411
Only other usage: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor_task/tasks_helper/misc.py